custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -104,6 +114,63 @@ def handle_exception(self, exception, repeat_record):
             _save_error_message(repeat_record.domain, repeat_record.payload_id, unicode(exception))
 
 
+@RegisterGenerator(NikshayHIVTestRepeater, 'case_json', 'JSON', is_default=True)
+class NikshayHIVTestPayloadGenerator(BasePayloadGenerator):
+    @property
+    def content_type(self):
+        return 'application/json'
+
+    def get_payload(self, repeat_record, person_case):
+        """
+        https://docs.google.com/document/d/1yUWf3ynHRODyVVmMrhv5fDhaK_ufZSY7y0h9ke5rBxU/edit#heading=h.hxfnqahoeag
+        """
+        episode_case = get_open_episode_case_from_person(person_case.domain, person_case.get_id)
+        episode_case_properties = episode_case.dynamic_case_properties()
+        person_case_properties = person_case.dynamic_case_properties()
+        properties_dict = {
+            "PatientID": episode_case_properties.get('nikshay_id'),
+            "HIVStatus": hiv_status.get(person_case_properties.get('hiv_status')),
+            "HIVTestDate": datetime.datetime.strptime(person_case_properties.get('hiv_test_date'),
+                                                      '%Y-%m-%d').strftime('%d/%m/%Y'),
+            "CPTDeliverDate": datetime.datetime.strptime(
+                person_case_properties.get('cpt_initiation_date', NIKSHAY_NULL_DATE), '%Y-%m-%d'
+            ).strftime('%d/%m/%Y'),
+            "ARTCentreDate": datetime.datetime.strptime(
+                person_case_properties.get('art_initiation_date', NIKSHAY_NULL_DATE), '%Y-%m-%d'
+            ).strftime('%d/%m/%Y'),
+            "InitiatedOnART": art_initiated.get(person_case_properties.get('art_initiated', 'no')),
+            "InitiatedDate": datetime.datetime.strptime(
+                person_case_properties.get('art_initiation_date', NIKSHAY_NULL_DATE), '%Y-%m-%d'
+            ).strftime('%d/%m/%Y'),
+            "Source": ENIKSHAY_ID,
+            "regby": repeat_record.repeater.username,
+            "password": repeat_record.repeater.password,
```

Comment:
Seems weird to be sending the password in the request body, but assuming this is a quirk of the Nikshay api?